### DISCUSSION TITLE
New project - Composer install on WSL fails on fakerphp/faker

### DISCUSSION THREAD

**Discussion by kinekt4** [laravel/framework#47226] [General]
I'm trying to install laravel 10 on `WSL` and I'm getting an error about failing to install `fakerphp/faker`.

In `composer.json` I can see
```
"fakerphp/faker": "^1.9.1",
```
But for some reason composer seems to lock faker at `v1.22.0`:
```
  - Locking fakerphp/faker (v1.22.0)
```
And then further down as the installation continues:
```
78/107 [====================>-------]  72%    Install of fakerphp/faker failed
    Install of nesbot/carbon failed
    Install of league/commonmark failed
    Install of laravel/framework failed
    Install of nikic/php-parser failed
    Install of psy/psysh failed
    Install of hamcrest/hamcrest-php failed
    Install of mockery/mockery failed
    Install of sebastian/diff failed
    Install of sebastian/comparator failed
```
I assume locking `faker` at `v1.22.0` is the problem.

I have tried several methods, all resulting in the same error:

1. `composer create-project  laravel/laravel`
2. `git clone git@github.com:laravel/laravel.git && cd laravel && composer install`

I don't have trouble installing laravel on a linux system (Fedora).

ps. Bringing `compose.lock` file from the fedora system to WSL did nothing to address this issue.

### Environment:
* OS: Windows 10 Pro
* WSL2: Ubuntu 22.04.2 LTS
* PHP: 8.1.2
* Composer: 2.2.6

**Comment by lvftavares** [comment]
I am having this trouble on WIndows 10 ... Any solution ?

**Comment by HanKienjd** [comment]
Any solution for this Error !

**Reply by a-h-abid** [reply]
Can you share the full error output from composer fails? It should have details on why it failed.

**Comment by ViNoS-ab** [comment]
same issue here
using docker
```
    Install of nesbot/carbon failed
    Install of laravel/framework failed
    Install of swagger-api/swagger-ui failed
    Install of facebook/php-business-sdk failed
    Install of fakerphp/faker failed
    Install of google/apiclient-services failed
    Install of google/protobuf failed
    Install of google/grpc-gcp failed
    Install of google/common-protos failed

In Filesystem.php line 333:
                                                                                                               
  Could not delete /var/www/html/script/vendor/composer/a5c5a754/briannesbitt-Carbon-afd4658/src/Carbon/Lang:  
                                                                                                               
```

**Reply by a-h-abid** [reply]
Could be file permission issue. Can you do `chmod -R ugo+w vendor/` and then try `composer install` ??

**Reply by ViNoS-ab** [reply]
tried, but not working

**Comment by mahadirazib** [comment]
Have you found the solution?

**Reply by ViNoS-ab** [reply]
switched to linux

**Reply by mahadirazib** [reply]
I am facing the same issue on windows. But solved somehow. I think, in my case, the problem was permission related.
tnx by the way.
